apps/elixir_ls_utils/priv/launch.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+# Actual launcher. This does the hard work of figuring out the best way
+# to launch the language server or the debugger.
+#
+
+# Running this script is a one-time action per project launch, so we opt for
+# code simplicity instead of performance. Hence some potentially redundant
+# moves here.
+
+# First order of business, see whether we can setup asdf-vm
+
+did_relaunch=$1
+
+asdf_vm="${HOME}/.asdf/asdf.sh"
+if test -f "${asdf_vm}"
+then
+  # asdf-vm does not support the plain posix shell. Figure out
+  # which one we need and relaunch ourselves with that.
+  case "${did_relaunch}" in
+    "")
+      if which bash >/dev/null
+      then
+        exec "$(which bash)" "$0" relaunch
+      elif which zsh >/dev/null
+      then
+        exec "$(which zsh)" "$0" relaunch
+      fi
+      ;;
+    *)
+      # We have an arg2, so we got relaunched. Therefore, e're running in
+      # a shell that can asdf-vm.
+      .  "${asdf_vm}"
+      ;;
+  esac
+fi
+
+# In case that people want to tweak the path, which Elixir to use, or
+# whatever prior to launching the language server or the debugger, we
+# give them the chance here. ELS_MODE will be set for
+# the really complex stuff. Use an XDG compliant path.
+
+els_setup="${HOME}/.config/elixir_ls/setup.sh"
```

Comment:
If you are trying to properly support XDG, this should be able to read from `$XDG_CONFIG_HOME`, or use `$HOME/.config` if that variable is not set.